pkg/runtime/runtime.go

```diff
@@ -229,3 +216,16 @@ func TemplateSpecApply[A any](info *Info) (*A, bool) {
 	spec, ok := info.TemplateSpec.ObjApply.(*A)
 	return spec, ok
 }
+
+func (i *Info) FindContainerByPodSetContainerName(psName, containerName string) (*int, *int) {
+	for psIdx, ps := range i.TemplateSpec.PodSets {
+		if ps.Name == psName {
+			for containerIdx, container := range ps.Containers {
+				if container.Name == containerName {
+					return &psIdx, &containerIdx
```

Comment:
Could this be `return &info.TemplateSpec.PodSets[psIdx].Containers[containerIdx]`?

If yes that would simplify things in the torch plugin.

Reply:
Yes, we can since your solution can return the pointer to callers.
Thank you for this suggestion.

Reply:
Done.